docker/Dockerfile

```diff
@@ -56,17 +56,17 @@ ARG VERSION="unknown"
 ARG GIT_COMMIT="unknown"
 RUN make cmds
 
-FROM ${CUDA_SAMPLE_IMAGE} AS sample-builder
-
-FROM nvcr.io/nvidia/cuda:12.9.0-base-ubi9
+# Download must-gather dependency: `kubectl`
+ARG TARGETARCH
+RUN OS_ARCH=${TARGETARCH/x86_64/amd64} && OS_ARCH=${OS_ARCH/aarch64/arm64} && \
+    curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/${OS_ARCH}/kubectl && \
+    chmod +x ./kubectl
 
-# Remove CUDA libs(compat etc) in favor of libs installed by the NVIDIA driver
-RUN dnf remove -y cuda-*
+FROM ${CUDA_SAMPLE_IMAGE} AS sample-builder
```

Comment:
One note about this is that using a build arg for this doesn't allow automatic updating of the docker images.